scuba/config.py

```diff
@@ -155,9 +155,12 @@ def override(self, node: yaml.nodes.Node) -> OverrideMixin:
         assert isinstance(obj, OverrideMixin)
         return obj
 
+    def not_yet_implemented(self, node: yaml.nodes.Node):
+        pass
```

Comment:
This implementation appears to just *ignore* the `!reference` tags, without raising an error or warning the user, simply returning `None` (implicitly). This seems like a *huge* problem to me.

Consider this example:
```yaml
.setup:
  script:
    - do-something-really-important

build:
  stage: build
  script:
    - !reference [.setup, script]
    - depends-on-important-stuff
```

The result (AFICT) is a script of `[None, "depends-on-important-stuff"]`.

I suspect that will crash somewhere later because of the `None` value.

If we pretend for a second that the None value is ignored, then the resulting script will be missing the first step, and executing anyway, despite the YAML author's intentions.

I don't see how this can be better than the current state of affairs (in which it just crashes).